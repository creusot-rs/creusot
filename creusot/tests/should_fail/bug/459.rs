extern crate creusot_contracts;
use creusot_contracts::*;

enum A {
    Cons(Box<A>),
    Nil,
}

impl DeepModel for A {
    type DeepModelTy = Int;
    #[logic]
    #[open]
    fn deep_model(self) -> Self::DeepModelTy {
        match self {
            A::Cons(a) => *a.deep_model() + 1,
            A::Nil => 0,
        }
    }
}
