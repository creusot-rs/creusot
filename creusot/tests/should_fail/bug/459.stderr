error: Mutually recursive functions: when calling `<A as creusot_contracts::DeepModel>::deep_model`...
  --> 459.rs:13:5
   |
13 |     fn deep_model(self) -> Self::DeepModelTy {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: then `<A as creusot_contracts::DeepModel>::deep_model` might call `<A as creusot_contracts::DeepModel>::deep_model` via the call to `creusot_contracts::std::boxed::<impl creusot_contracts::DeepModel for std::boxed::Box<T, A>>::deep_model`.
  --> 459.rs:15:28
   |
15 |             A::Cons(a) => *a.deep_model() + 1,
   |                            ^^^^^^^^^^^^^^

error: aborting due to 1 previous error

